code/modules/modular_computers/file_system/programs/statusdisplay.dm

```diff
@@ -12,44 +12,79 @@
 	usage_flags = PROGRAM_ALL
 	available_on_ntnet = FALSE
 
-	var/upper_text
-	var/lower_text
+	var/upper_text = ""
+	var/lower_text = ""
 
-/datum/computer_file/program/status/proc/SendSignal()
+/**
+ * Post status display radio packet.
+ * Arguments:
+ * * command - the status display command
+ * * data1 - the data1 value, as defined by status displays
+ * * data2 - the data2 value, as defined by status displays
+ */
+/datum/computer_file/program/status/proc/post_status(command, data1, data2)
 	var/datum/radio_frequency/frequency = SSradio.return_frequency(FREQ_STATUS_DISPLAYS)
-
 	if(!frequency)
 		return
 
-	var/datum/signal/status_signal = new(list("command" = "message"))
+	var/datum/signal/status_signal = new(list("command" = command))
+	switch(command)
+		if("message")
+			status_signal.data["top_text"] = data1
+			status_signal.data["bottom_text"] = data2
+		if("alert")
+			status_signal.data["picture_state"] = data1
+
+	frequency.post_signal(src, status_signal)
 
-	status_signal.data["msg1"] = reject_bad_text(upper_text || "", MAX_STATUS_LINE_LENGTH)
-	status_signal.data["msg2"] = reject_bad_text(lower_text || "", MAX_STATUS_LINE_LENGTH)
+/**
+ * Post a message to status displays
+ * Arguments:
+ * * upper - Top text
+ * * lower - Bottom text
+ */
+/datum/computer_file/program/status/proc/post_message(upper, lower)
+	post_status("message", upper, lower)
 
-	frequency.post_signal(computer, status_signal)
+/**
+ * Post a picture to status displays
+ * Arguments:
+ * * picture - The picture name
+ */
+/datum/computer_file/program/status/proc/post_picture(picture)
+	var/static/obj/machinery/computer/communications/comm_prototype = /obj/machinery/computer/communications
```

Comment:
I really don't like how this is done, I think these could be vars on the status displays (?) or preferably just a global list

Reply:
100%. Changed to use a GLOB.